Quadriga/src/main/java/edu/asu/spring/quadriga/rest/NetworkRestController.java

```diff
@@ -663,5 +676,67 @@
         return new ResponseEntity<String>(res.toString(), httpHeaders, HttpStatus.OK);
 
     }
+    
+    
+    /**
+     * Rest interface to search concept in list of projects
+     * http://<<URL>:<PORT>>/quadriga/rest/sites/networks/search/{conceptId}/{projectIds}
+     * networks
+     * 
+     * @param conceptId
+     * @param projectIds
+     * @param response
+     * @param accept
+     * @param principal
+     * @return status
+     * @throws RestException
+     */
+    
+    @RequestMapping(value = "rest/sites/networks/search", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> getSearchTransformedNetwork(@RequestParam("conceptId") String conceptId, @RequestParam("projectIds") List<String> projectIds,
+            HttpServletResponse response, String accept, Principal principal, HttpServletRequest req) throws RestException {
+    	
+    	List<IProject> projectList = new ArrayList<>();
+    	
+    	try {    		
+    		for(String projectId : projectIds){
+        		IProject project = projectManager.getProjectDetails(projectId);
+        		if(project!=null){
+        			projectList.add(project);
+        		}
+        	}
+    		
+    		if(projectList.size()==0){
+                String errorMsg = errorMessageRest.getErrorMsg("Projects doesn't exist");
+                return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        	}
 
+    		ITransformedNetwork transformedNetwork = transformationManager.getSearchTransformedNetworkMultipleProjects(
+        			projectIds, conceptId);	
+    		
+    		VelocityEngine engine = restVelocityFactory.getVelocityEngine(req);
+            engine.init();
+            Template template = engine.getTemplate("velocitytemplates/transformationdetails.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("nodeList", new ArrayList<Node>(transformedNetwork.getNodes().values()));
+            context.put("linkList", transformedNetwork.getLinks());
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.setContentType(MediaType.APPLICATION_XML);
+            return new ResponseEntity<String>(writer.toString(), httpHeaders, HttpStatus.OK);
+        } catch (ResourceNotFoundException e) {
+            throw new RestException(404, e);
+        } catch (ParseErrorException e) {
+            throw new RestException(500, e);
+        } catch (MethodInvocationException e) {
+            throw new RestException(500, e);
+        } catch (QuadrigaStorageException e) {
+            throw new RestException(500, e);
+        } catch (QuadrigaAccessException e) {
+            throw new RestException(403, e);
+        } catch (Exception e) {
+            throw new RestException(500, e);
+        }

```

Comment:
your try/catch blocks should be around the code that's throwing it, not all of it. You need to handle the exceptions so that they make sense. E.g. just because, there is an issue with one project, doesn't have to mean that none can't be retrieved.